plugins/jenkins/src/components/useJobRuns.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useEffect, useState } from 'react';
+import useAsyncRetry from 'react-use/lib/useAsyncRetry';
+import { jenkinsApiRef } from '../api';
+import { errorApiRef, useApi } from '@backstage/core-plugin-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { getCompoundEntityRef } from '@backstage/catalog-model';
+
+export enum ErrorType {
+  CONNECTION_ERROR,
+  NOT_FOUND,
+}
+
+export function useJobRuns(jobFullName: string) {
+  const { entity } = useEntity();
+  const api = useApi(jenkinsApiRef);
+  const errorApi = useApi(errorApiRef);
+
+  const [total, setTotal] = useState(0);
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+
+  const [error, setError] = useState<{
+    message: string;
+    errorType: ErrorType;
+  }>();
+
+  const { loading, value: jobRuns } = useAsyncRetry(async () => {
+    try {
+      const jobBuilds = await api.getJobBuilds({
+        entity: getCompoundEntityRef(entity),
+        jobFullName,
+      });
+      return jobBuilds;
+    } catch (e) {
+      const errorType = e.notFound
+        ? ErrorType.NOT_FOUND
+        : ErrorType.CONNECTION_ERROR;
+      setError({ message: e.message, errorType });
+      throw e;
+    }
+  }, [api, errorApi, entity]);
+
+  useEffect(() => {
+    if (jobRuns) setTotal(jobRuns.builds.length);
```

Comment:
Why do you store the total in a separate state? You can just return them directly below as `total: jobRuns.builds.length`

Reply:
ok, removing this as i can use the default value and jobRuns to eliminate any need of this prop